datetime_formatter/models.py

```diff
@@ -0,0 +1,60 @@
+# -*- encoding: utf-8 -*-
+from openerp import api, fields, models
+from openerp.tools import (DEFAULT_SERVER_DATE_FORMAT,
+                           DEFAULT_SERVER_TIME_FORMAT)
+
+# Available modes for :param:`.ResLang.datetime_formatter.template`
+MODE_DATETIME = "MODE_DATETIME"
+MODE_DATE = "MODE_DATE"
+MODE_TIME = "MODE_TIME"
+
+
+class ResLang(models.Model):
+    _inherit = "res.lang"
+
+    @api.model
+    def datetime_formatter(self, value, lang=None, template=MODE_DATETIME,
+                           separator=" "):
+        """Convert a datetime field to lang's default format.
+
+        :type value: `str` or `datetime.datetime`
+        :param value:
+            Datetime that will be formatted to the user's preferred format.
+
+        :param str lang:
+            Language code, formatted like "en_US". If not set, the program will
+            try to guess the best match.
+
+        :param str template:
+            Will be used to format :param:`value`. If it is one of the special
+            constants :const:`MODE_DATETIME`, :const:`MODE_DATE` or
+            :const:`MODE_TIME`, it will use the :param:`lang`'s default
+            template for that mode.
+
+        :param str separator:
+            Only used when :param:`template` is :const:`MODE_DATETIME`, as the
+            separator between the date and time parts.
+        """
+        # Get the correct lang
+        if not lang:
+            lang = (self.env.context.get("lang") or
+                    self.env.user.lang or
+                    self.search([("active", "=", True)], limit=1).code)
```

Comment:
you might want to check before if self.ids is truthy, then you were called on a browse record, and in this case, simply use self[:1][0]. This way, we can say things like `my_object.lang.datetime_formatter(mydate)` and get it in the object's languge